File: Sorting Algorithms/QuickSort.java
public class QuickSort
{
    private static int partition(int[] arr, int low, int high)
    {
        int pivot = arr[high];

        int i = low - 1;

        for(int j = low; j < high; j++)
        {
            if(arr[j] < pivot)
            {
                i += 1;

                //swap arr[i] and arr[j]
                int temp = arr[j];
                arr[j] = arr[i];
                arr[i] = temp;
            }
        }
        //swap arr[i+1] and pivot
        //int temp = pivot;
        arr[high] = arr[i+1];
        arr[i+1] = pivot;

        return i+1;
    }

    private static void sort(int[] arr, int low, int high)
    {
        if (low < high)
        {
            int pi = partition(arr, low, high);

            sort(arr, low, pi - 1);
            sort(arr, pi + 1, high);
        }
    }
    
    private static void print(int[] arr)
    {
        for(int i : arr)
            System.out.print(i + " ");

        System.out.println();
    }
    public static void main(String[] args) {
        int[] arr = {10, 7, 8, 9, 1, 5};

        sort(arr, 0,arr.length -1);
        print(arr);
    }
}

File: Sorting Algorithms/Algorithms/QuickSort.py
__author__="Kalpak Seal"

def partition(arr, low, high):
        pivot = arr[high]
        i = low - 1

        for j in range(low, high):
                if arr[j] <= pivot:
                        i += 1
                        arr[i], arr[j] = arr[j], arr[i]

        arr[i+1], arr[high] = arr[high], arr[i+1]
        return (i+1)

def quick_sort(arr, low, high):
        if low < high:
                pi = partition(arr, low, high)
                quick_sort(arr, low, pi - 1)
                quick_sort(arr, pi + 1, high)

if __name__=='__main__':
        arr = [int(x) for x in input('Enter the numbers to sort').split()]
        n = len(arr)
        quick_sort(arr, 0, n-1)
        print (arr)

